backend/authentication/models.py

```diff
@@ -59,6 +59,11 @@ class User(AbstractUser):
 
     badges = ArrayField(models.CharField(max_length=30), default=empty_list)
 
+    followings = models.ManyToManyField('self')
+    followers = models.ManyToManyField('self')
+    blocked = models.ManyToManyField('self')
+    blocked_by = models.ManyToManyField('self')
+
```

Comment:
Just curious because I have never used a ManyToManyField, what does the 'self' argument mean?

Reply:
Since the all those are also `User` models, and this code is in the definition of the `User` model, `self` argument is used to refer to itself.

For instance, if I were to add a `User` as `ForeignKey`, I would still use `self` in the following way: 
`models.ForeignKey('self', on_delete=CASCADE)`

But since we need lists here, we are using `ManyToManyField`s and no option for what happens on delete.

At first I tried using `ArrayField(models.ForeignKey('self', on_delete=CASCADE), default=empty_list)`, but apparently that is not how it works. For foreign keys, `ManyToManyField`s are used.